bundle/config/mutator/run_as.go

```diff
@@ -23,12 +27,120 @@ func (m *setRunAs) Name() string {
 	return "SetRunAs"
 }
 
+// Resources that satisfy one of the following conditions:
+//  1. Allow to set run_as for the resources to a different user from the current
+//     deployment user. For example, jobs.
+//  2. Does not make sense for these resources to run_as a different user. We do not
+//     have plans to add platform side support for `run_as` for these resources.
+//     For example, experiments or registered models.
+var allowListForRunAsOther = []string{"jobs", "models", "registered_models", "experiments"}
+
+// Resources that do not allow setting a run_as identity to a different user but
+// have plans to add platform side support for `run_as` for these resources at
+// some point in the future. For example, pipelines, model serving endpoints or lakeview dashboards.
+//
+// We expect the allow list and the deny list to form mutually exclusive and exhaustive
+// sets of all resource types that are supported by DABs.
+var denyListForRunAsOther = []string{"pipelines", "model_serving_endpoints"}
+
+type errorUnsupportedResourceTypeForRunAs struct {
+	resourceType  string
+	resourceValue dyn.Value
+	currentUser   string
+	runAsUser     string
+}
+
+// TODO(6 March 2024): This error message is big. We should split this once
+// diag.Diagnostics is ready.
+// TODO(6 March 2024): Link the docs page describing run_as semantics in the error below
+// once the page is ready.
+func (e errorUnsupportedResourceTypeForRunAs) Error() string {
+	return fmt.Sprintf("%s are not supported when the current deployment user is different from the bundle's run_as identity. Please deploy as the run_as identity. List of supported resources: [%s]. Location of the unsupported resource: %s. Current identity: %s. Run as identity: %s", e.resourceType, strings.Join(allowListForRunAsOther, ", "), e.resourceValue.Location(), e.currentUser, e.runAsUser)
```

Comment:
I don't think it matters in the error message which resources support top level run_as

Reply:
The idea here was to give all the information to the users upfront, instead of them adding a resource to their DAB, and then finding out it's not supported in the way the are trying to use it.

I'm not strongly opinionated about this though, incase you believe the error message should be shorter here.

Reply:
Removed the list of supported resources.